enhancements/builds/disable-builder-sa.md

```diff
@@ -0,0 +1,362 @@
+  so that I can provide my own RBAC for builds in the “golden path” namespace
+  template for dev teams.
+- As an information security officer, I want to disable the builder service
+  account as part of our process to limit access to the OCP Build system on
+  production/application clusters so that only service accounts related to
+  applications are deployed, and they have the minimum permissions necessary.
+- As a software architect/platform engineer, I want to change the default
+  service account used for builds so I can customize its permissions.
+- As a product manager, I want to know how many OpenShift clusters are
+  disabling the builder service account so that I can understand the impact of
+  this feature.
+
+### Goals
+
+- On existing clusters that have the `Build` capability enabled, cluster admins
+  can disable the creation of the `builder` service account.
+- On existing clusters that the `builder` and service accounts are disabled,
+  cluster admins can enable creation of these service accounts in every
+  namespace.
+- When the `builder` service account is disabled, existing `builder` service
+  accounts already created in namespaces should remain intact.
+- When the `builder` service account is disabled, manual deletion of existing
+  `builder` service accounts should not lead to recreation of these service
+  accounts.
+
+
+### Non-Goals
+
+- Cleaning up and deleting existing builder service accounts in namespaces
+  after the SA generation controller is disabled.
+- Enable/disable the `deployer` service account. This will require a different
+  implementation/API.
+- Disable the `BuildConfig` API and controllers on clusters that have the `Build`
+  capability enabled at install time, or turn on the Build capability after
+  installation.
+- Refactoring “system:*” bootstrap roles and rolebindings related to BuildConfigs.
+- Add Service Accounts as a `buildDefault`/`buildOverride` feature.
+- Fine tune the RBAC of the generated `builder` service account.
+
+
+## Proposal
+
+The general idea is to expose a new cluster-level configuration that controls
+if the `builder` service account is created.
+
+### Workflow Description
+
+1. The cluster is created or upgaded with the `Build` capability enabled.
+2. On install or upgrade with the `Build` capability enabled, the `Build`
+   cluster configuration CRD will be installed. The CRD will have a new field
+   to declare that the `builder` service account should be generated:
+
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Build
+   spec:
+     buildDefaults:
+       ...
+     buildOverrides:
+       ...
+     builderServiceAccount: Generate   
+   ```
+
+3. The cluster administrator edits the cluster build system configuration to
+   disable the generation of the builder serice account.
+
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Build
+   spec:
+     ...
+     builderServiceAccount: Disable
```

Comment:
isn't it easier to just call this field `builderServiceAccountDisabled: true` and get rid of the "Generate" ? in that case you don't have to default, etc.

Reply:
The general guideline is against boolean fields, especially when we envision additional behaviors, just like Adam explained above.